nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/admin/IdmService.java

```diff
@@ -184,6 +186,14 @@ private boolean setFromStdin() {
         } catch (Exception e) {
             logger.fine("Stdin isn't behaving, ignoring it ..." + e.getMessage());
             return false;
+        } finally {
+            try {
+                System.in.close();
```

Comment:
I understand that
> Closing a previously closed stream has no effect.

But wouldn't `stdin` be closed already by [since now] try-with-resources with closing `br`?

> BufferedReader.close
> Closes the stream and releases any system resources associated with it

> InputStreamReader.close
> Closes the stream and releases any system resources associated with it.

Reply:
> But wouldn't `stdin` be closed already by [since now] try-with-resources with closing `br`?

Answer is: yes, it would be closed.

I'd like to see that `finally` block to be removed.

Reply:
I would say so, but honestly I gave up after another "Cannot delete the directory" error on Windows. Maybe I made some mistake when I was testing that, like using old build, not sure now. Or maybe there was yet another change causing the problem. I will come back to this soon as I know yet about one place with similar issue.